core/isolate.rs

```diff
@@ -323,22 +335,19 @@ impl Isolate {
         source.as_ptr(),
       )
     };
-    if let Some(err) = self.last_exception() {
-      return Err(err);
-    }
-    Ok(())
+    self.check_last_exception()
   }
 
-  fn last_exception(&self) -> Option<JSError> {
+  fn check_last_exception(&self) -> Result<(), ErrBox> {
```

Comment:
Please add a triple slash comment saying the ErrBox should always be downcast to CoreJsError.